lib/v2/showstart/service.js

```diff
@@ -0,0 +1,44 @@
+const qs = require('query-string');
+const got = require('@/utils/got');
+const cheerio = require('cheerio');
+
+const host = 'https://www.showstart.com';
+
+async function fetchShowRSS({ keyword, showStyle, cityCode }) {
+    const url = `${host}/event/list?${qs.stringify(
+        {
+            pageNo: 1,
+            pageSize: 30,
+            cityCode: cityCode || '',
+            showStyle: showStyle || '',
+            keyword: keyword || '',
+        },
+        {
+            skipEmptyString: true,
+        }
+    )}`;
+
+    const respond = await got.get(url);
+    const $ = cheerio.load(respond.data);
+
+    const items = $('.list-box .show-item')
+        .toArray()
+        .map((item) => ({
+            title: $(item).find('.title').text(),
+            link: host + $(item).attr('href'),
+            description: $(item).find('.addr').text(),
+        }));
+
+    return {
+        $,
+        data: {
+            title: '秀动网',
+            link: host,
+            item: items,
+        },
+    };
+}
```

Comment:
Please use their web API at `/api/web/activity/list` instead.

<details><summary>Quick Snippets</summary>

<details><summary>event.js</summary>

```js
const { post, getAccessToken } = require('./utils');

module.exports = async (ctx) => {
    const accessToken = await getAccessToken();

    const result = await post('/web/activity/list', accessToken, {
        pageNo: '1',
        pageSize: '20',
        cityCode: '', // ctx.params.cityCode
        activityIds: '',
        coupon: '',
        keyword: '',
        organizerId: '',
        performerId: '',
        showStyle: '',
        showTime: '',
        showType: '',
        siteId: '',
        sortType: '',
        themeId: '',
        timeRange: '',
        tourId: '',
        type: '',
        tag: '',
    });

    console.log(JSON.stringify(result));
};
```
</details>

<details><summary>utils.js</summary>

```js
const got = require('@/utils/got');
const md5 = require('@/utils/md5');

const uuid = (length = 20) => {
    const e = '0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz' + Date.now();
    const r = [];
    for (let i = 0; i < length; i++) {
        r.push(e.charAt(Math.floor(Math.random() * e.length)));
    }
    return r.join('');
};

const cookieMap = new Map([['token', uuid(32).toLowerCase()]]);

const devioceInfo = {
    vendorName: '',
    deviceMode: '',
    deviceName: '',
    systemName: '',
    systemVersion: '',
    cpuMode: ' ', // Note the space
    cpuCores: '',
    cpuArch: '',
    memerySize: '',
    diskSize: '',
    network: '',
    resolution: '1920*1080',
    pixelResolution: '',
};

const getAccessToken = async () => {
    const { result } = await post('/waf/gettoken');
    cookieMap.set('accessToken', result.accessToken.access_token);
    cookieMap.set('idToken', result.idToken.id_token);
    return cookieMap.get('accessToken');
};

const post = async (requestPath, accessToken = md5(Date.now().toString()), payload) => {
    const traceId = uuid(32) + Date.now();

    const { data: response } = await got.post(`https://www.showstart.com/api${requestPath}`, {
        headers: {
            cdeviceinfo: encodeURIComponent(JSON.stringify(devioceInfo)),
            cdeviceno: cookieMap.get('token'),
            cookie: [...cookieMap.entries()].map(([key, value]) => `${key}=${value}`).join('; '),
            crpsign: md5(accessToken + 'web' + cookieMap.get('token') + (payload ? JSON.stringify(payload) : '') + requestPath + '999web' + traceId),
            crtraceid: traceId,
            csappid: 'web',
            cterminal: 'web',
            cusat: accessToken,
            cusid: '',
            cusit: '',
            cusname: '',
            cusut: '',
            cversion: '999',
        },
        json: payload,
    });

    return response;
};

module.exports = {
    post,
    getAccessToken,
    uuid,
};
```
</details>

</details>

Reply:
You can also obtain a mapping of `cityCode`, `showStyle` using
```js
await post('/web/activity/list/params', accessToken)
```

Reply:
Thanks a lot, It's really helpful, I will change the code later.